src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs

```diff
@@ -728,6 +730,51 @@ public async Task<bool> WriteConst(LiteralExpressionSyntax constValue, MonoSDBHe
             return false;
         }
 
+        public bool WriteNumber(object number, bool convertToNegative=false)
+        {
+            int coeff = convertToNegative ? -1 : 1;
+            switch (number)
+            {
+                case double d:
+                    Write(ElementType.R8, d * coeff);
+                    break;
+                case float f:
+                    Write(ElementType.R4, f * coeff);
+                    break;
+                case long l:
+                    Write(ElementType.I8, l * coeff);
+                    break;
+                case ulong ul:
+                    Write(ElementType.U8, ul);
+                    break;
+                case byte b:
+                    Write(ElementType.U1, (int)b);
+                    break;
+                case sbyte sb:
+                    Write(ElementType.I1, (uint)sb);
+                    break;
+                case ushort us:
+                    Write(ElementType.U2, (int)us);
+                    break;
+                case short s:
+                    Write(ElementType.I2, (uint)s);
```

Comment:
Isn't `* coeff` needed here too?

Reply:
And add the test cases for all the types which support a negative number.

Reply:
Good catch